lib/src/main/kotlin/MCTSSolver.kt

```diff
@@ -0,0 +1,283 @@
+import kotlin.math.ln
```

Comment:
Is this class needed anymore? Compare this with the StatefulSolver and see if there are anything different. Afterwards, we should probably remove this.